src/main/java/uk/gov/hmcts/ccd/domain/model/common/CaseReferenceUtils.java

```diff
@@ -11,4 +13,20 @@ public static String formatCaseReference(String caseReference) {
         }
         return caseReference;
     }
+
+    public static boolean checkRegex(final String value) {
+        final String regularExpression = "(?:^[0-9]{16}$|^\\d{4}-\\d{4}-\\d{4}-\\d{4}$)";
+        return value.matches(regularExpression);
+    }
+
+    public static Optional<String> getFormatCaseReference(Optional<String> caseReference) {
```

Comment:
why you passing the Optional? in terms of API it's better if you pass the String, and it's the code that calls the that deals with the Optional.
Otherwise every code that calls this method will need to wrap the string in an Optional